greenpeace/planet4-master-theme
# PLANET-6257: New country selector design implementation

Ref: https://jira.greenpeace.org/browse/PLANET-6257

Implementation of the new Country Selector as described in the design system https://p4-designsystem.greenpeace.org/05f6e9516/v/0/p/16a899-footer

## Feature

__This is implemented behind a feature toggle__, so that every child-theme has time to adapt. 
In the admin panel, section _Planet 4 > Features_:
![Screenshot from 2021-11-24 12-37-18](https://user-images.githubusercontent.com/617346/143231467-db69181d-d372-4d4f-a196-4403f76f7101.png)


This will remove the country selector from the navigation bar and add it to the footer.
![Screenshot from 2021-11-04 19-51-48](https://user-images.githubusercontent.com/617346/140401875-65c44eeb-9149-4d1c-b526-90bf821f153b.png)
## Modifications

- `h3` tag was removed, the list structure and labels should be enough to express the meaning of elements
- Links have an `hreflang` attribute to indicate the main language of the URL
- JS script was dropped in favor of a backend generated template (cf `templates/footer_country_selector.twig`)

## For child themes extension

- A twig template [`templates/footer_country_selector.twig`](https://github.com/greenpeace/planet4-master-theme/pull/1471/files#diff-26a6ff6c33f468b55994daf5fd0006c20cef6073c1f06feec8200a30fff33c8a) is included in the `templates/footer.twig`.  
  Its content is generated from a backend variable [`countries_by_initials`](https://github.com/greenpeace/planet4-master-theme/pull/1471/files#diff-d5a71432d6e94abc1260a448df306d8dfc76d15fa4047646a4d133c38cc131d7R558-R561), coming from the json file `templates/countries.json`.
- The country selector colors should adapt to the footer colors by default.

![Screenshot from 2021-11-17 20-32-52](https://user-images.githubusercontent.com/617346/142271778-bd1ac947-c953-4f6e-8022-63e01f1e3454.png)

## Test

- Activate the feature in Planet4 > Features "Enable  Country selector design"
- Check the Country selector in the footer of the frontend - it should also adapt to campaigns colors, and doesn't appear in minimal navigation mode
